onelogin_rules/onelogin_unusual_login.py

```diff
@@ -0,0 +1,60 @@
+import json
+import requests
+from panther_oss_helpers import get_string_set, put_string_set
+
+
+def rule(event):
+    # Pre-filter to save compute time where possible. event_type_id = 5 is login events.
+    if event.get('event_type_id') != 5 or event.get('ipaddr') is None:
+        return False
+
+    # Lookup geo-ip data via API call
+    url = 'https://ipinfo.io/' + event['ipaddr'] + '/geo'
+
+    # Skip API call if this is a unit test
+    if __name__ == 'PolicyApiTestingPolicy':
+        resp = lambda: None
+        setattr(resp, 'status_code', 200)
+        setattr(resp, 'text', event['api_data'])
+    else:
+        resp = requests.get(url)
+
+    if resp.status_code != 200:
+        # Could raise an exception here for ops team to look into
```

Comment:
Should we print or log a warning, then?

Reply:
Hmm good suggestion, I know everything that gets printed in the rules engine gets output to CloudWatch, but I'm not sure what formatting would be needed for our CloudWatch alarms to pick up on it. My main thought for this comment was that once we have better error reporting on the rules engine (i.e. straight to users, not to ops team) then we could raise an exception here. I will look into it some more.